pkg/cache/v3/linear.go

```diff
@@ -277,9 +308,71 @@ func (cache *LinearCache) CreateWatch(request *Request, value chan Response) fun
 }
 
 func (cache *LinearCache) CreateDeltaWatch(request *DeltaRequest, state stream.StreamState, value chan DeltaResponse) func() {
+	cache.mu.Lock()
+	defer cache.mu.Unlock()
+
+	response := respondDeltaLinear(request, value, state, cache, cache.log)
```

Comment:
Nice, maybe it makes a bit more sense to move that respondDeltaLinear function into this life? What do you think

Reply:
Yep, I can move both `respondDeltaLinear` and `respondDeltaSnapshot` to the corresponding files after we agree on remaining implementation details.